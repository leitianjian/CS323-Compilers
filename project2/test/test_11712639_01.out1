Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          ID: ST
          LC
          DefList (2)
            Def (2)
              Specifier (2)
                TYPE: int
              DecList (2)
                Dec (2)
                  VarDec (2)
                    VarDec (2)
                      VarDec (2)
                        ID: arr
                      LB
                      INT: 4
                      RB
                    LB
                    INT: 2
                    RB
                COMMA
                DecList (2)
                  Dec (2)
                    VarDec (2)
                      VarDec (2)
                        ID: arr1
                      LB
                      INT: 4
                      RB
                  COMMA
                  DecList (2)
                    Dec (2)
                      VarDec (2)
                        ID: val1
              SEMI
          RC
      SEMI
    ExtDefList (5)
      ExtDef (5)
        Specifier (5)
          TYPE: int
        FunDec (5)
          ID: test
          LP
          RP
        CompSt (5)
          LC
          DefList (6)
            Def (6)
              Specifier (6)
                StructSpecifier (6)
                  STRUCT
                  ID: ST
              DecList (6)
                Dec (6)
                  VarDec (6)
                    ID: s
              SEMI
            DefList (7)
              Def (7)
                Specifier (7)
                  TYPE: int
                DecList (7)
                  Dec (7)
                    VarDec (7)
                      VarDec (7)
                        ID: arr2
                      LB
                      INT: 3
                      RB
                SEMI
          StmtList (8)
            Stmt (8)
              Exp (8)
                Exp (8)
                  Exp (8)
                    Exp (8)
                      ID: arr2
                    LB
                    Exp (8)
                      INT: 1
                    RB
                  LB
                  Exp (8)
                    INT: 1
                  RB
                ASSIGN
                Exp (8)
                  INT: 3
              SEMI
            StmtList (9)
              Stmt (9)
                Exp (9)
                  Exp (9)
                    Exp (9)
                      Exp (9)
                        ID: s
                      DOT
                      ID: arr2
                    LB
                    Exp (9)
                      INT: 3
                    RB
                  ASSIGN
                  Exp (9)
                    INT: 2
                SEMI
              StmtList (10)
                Stmt (10)
                  Exp (10)
                    Exp (10)
                      Exp (10)
                        Exp (10)
                          ID: s
                        DOT
                        ID: arr
                      LB
                      Exp (10)
                        INT: 2
                      RB
                    ASSIGN
                    Exp (10)
                      ID: arr2
                  SEMI
                StmtList (11)
                  Stmt (11)
                    Exp (11)
                      Exp (11)
                        Exp (11)
                          Exp (11)
                            Exp (11)
                              ID: s
                            DOT
                            ID: arr1
                          LB
                          Exp (11)
                            INT: 2
                          RB
                        LB
                        Exp (11)
                          INT: 1
                        RB
                      ASSIGN
                      Exp (11)
                        INT: 1
                    SEMI
                  StmtList (12)
                    Stmt (12)
                      Exp (12)
                        Exp (12)
                          Exp (12)
                            Exp (12)
                              ID: s
                            DOT
                            ID: arr1
                          LB
                          Exp (12)
                            INT: 1
                          RB
                        ASSIGN
                        Exp (12)
                          INT: 0
                      SEMI
                    StmtList (13)
                      Stmt (13)
                        Exp (13)
                          Exp (13)
                            Exp (13)
                              ID: s
                            DOT
                            ID: val1
                          ASSIGN
                          Exp (13)
                            FLOAT: 3.0
                        SEMI
                      StmtList (14)
                        Stmt (14)
                          RETURN
                          Exp (14)
                            INT: 1
                          SEMI
          RC
