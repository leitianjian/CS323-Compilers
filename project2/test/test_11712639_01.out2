Error type 10 at Line 8: applying indexing operator ([...]) on non-array type variables
Error type 5 at Line 8: unmatching type on both sides of assignment
Error type 13 at Line 9: accessing an undefined structure member
Error type 10 at Line 9: applying indexing operator ([...]) on non-array type variables
Error type 5 at Line 9: unmatching type on both sides of assignment
Error type 10 at Line 11: applying indexing operator ([...]) on non-array type variables
Error type 5 at Line 11: unmatching type on both sides of assignment
Error type 5 at Line 13: unmatching type on both sides of assignment
