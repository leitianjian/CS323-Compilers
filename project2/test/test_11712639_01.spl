struct ST{
    int arr[4][2], arr1[4], val1;
};

int test(){
    struct ST s;
    int arr2[3];
    arr2[1][1] = 3;
    s.arr2[3] = 2;
    s.arr[2] = arr2;
    s.arr1[2][1] = 1;
    s.arr1[1] = 0;
    s.val1 = 3.0;
    return 1;
}