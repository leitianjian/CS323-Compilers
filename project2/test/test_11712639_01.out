Error type 10 at line 8: applying indexing operator ([...]) on non-array type variables
Error type 14 at line 9: accessing an undefined structure member
Error type 10 at line 11: applying indexing operator ([...]) on non-array type variables
Error type 5 at line 13: unmatching types on both sides of assignment operator
